### PR TITLE
adds the same mypy strictness as scadmin other than strict optional, which I will work on in a followup

### PR DESCRIPTION
**Pull request by Pastromhaug**

### CODE REVIEW

File: stopcovid/dialog/engine.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def execute(
         # a chain of responsibility. Each handler can handle the current command and return an
         # event list. A handler can also NOT handle an event and return None, thereby leaving it
         # for the next handler.
-        for handler in [
+        handlers: List[Callable[[DialogState, Dict[str, Any]], Optional[List[DialogEvent]]]] = [
```
Comment (jeffreysilver) [inline stopcovid/dialog/engine.py:140]: 😻

File: stopcovid/dialog/models/state.py [hunk 2]
```diff
@@ -31,13 +31,12 @@ class UserProfile(pydantic.BaseModel):
...
-    def __str__(self):
+    def __str__(self) -> str:
         return f"lang={self.language}, validated={self.validated}, " f"name={self.name}"
 
     @pydantic.validator("language", pre=True, always=True)
-    def set_language(cls, value):
-        if value is not None:
-            return value.lower()[:2]
+    def set_language(cls, value: Optional[str]) -> Optional[str]:
+        return value.lower()[:2] if value else None
```
Comment (jeffreysilver) [inline stopcovid/dialog/models/state.py:39]: we're changing the behavior for empty strings here.... is that going to be an issue?

File: stopcovid/dialog/registration.py [hunk 3]
```diff
@@ -22,13 +22,13 @@ class CodeValidationPayload(pydantic.BaseModel):
 
 class RegistrationValidator(ABC):
     @abstractmethod
-    def validate_code(self, code) -> CodeValidationPayload:
+    def validate_code(self, code: str) -> CodeValidationPayload:
         pass
 
 
 class DefaultRegistrationValidator(RegistrationValidator):
     @functools.lru_cache(maxsize=1024)
```
Comment (jeffreysilver) [inline stopcovid/dialog/registration.py:30]: forgot we were doing this. its so cool

Comment (jeffreysilver) [reply stopcovid/dialog/registration.py:30]: curious about how many cache hits we actually get, though

**Review by jeffreysilver — APPROVED** [review]

File: stopcovid/dialog/models/state.py (hunk 2 above)
Comment (Pastromhaug) [reply stopcovid/dialog/models/state.py:39]: good call. No reason to have functionality changes in this PR. Will fix.

**Pull request merged**